pisi/atomicoperations.py

```diff
@@ -270,38 +270,39 @@ def check_operation(self):
             ipkg = self.installdb.get_package(pkg.name)
             (iversion_s, irelease_s, ibuild) = self.installdb.get_version(pkg.name)
 
-            # determine if same version
+            # determine if same distribution release
             if pkg.release == irelease_s:
                 if self.ask_reinstall:
-                    if not ctx.ui.confirm(_("Re-install same version package?")):
+                    if not ctx.ui.confirm(_("Re-install same distribution release of package?")):
                         raise Error(_("Package re-install declined"))
                 self.operation = REINSTALL
             else:
-                pkg_version = pisi.version.make_version(pkg.version)
-                iversion = pisi.version.make_version(iversion_s)
-
-                pkg_release = int(pkg.release)
-                irelease = int(irelease_s)
+                #pkg_version = pisi.version.make_version(pkg.version)
+                #ctx.ui.debug(f"New upstream version after version parsing: {pkg_version}")
+                #iversion = pisi.version.make_version(iversion_s)
+                #ctx.ui.debug(f"Installed version after version parsing: {iversion}")
+                #pkg_release = int(pkg.release)
+                #irelease = int(irelease_s)
 
                 # is this an upgrade?
                 # determine and report the kind of upgrade: version, release
-                try:
-                    if pkg_version > iversion:
-                        ctx.ui.info(_("Upgrading to new upstream version"))
-                        self.operation = UPGRADE
-                except TypeError as e:
-                    ctx.ui.error(_("Malformed package version, falling back to distribution release"))
-                    ctx.ui.debug(str(e))
+                #try:
+                #    if pkg_version > iversion:
+                #        ctx.ui.info("Upgrading to new upstream version")
+                #        self.operation = UPGRADE
+                #except TypeError as e:
+                #    ctx.ui.error(f"Comparing new upstream version '{pkg_version}' to installed version '{iversion}' failed, falling back to distribution release comparison (Add '--debug' to see full traceback)")
+                #    ctx.ui.debug(str(e))
                 if pkg_release > irelease:
-                    ctx.ui.info(_("Upgrading to new distribution release"))
+                    ctx.ui.info(_("Upgrading to new upstream distribution release"))
```

Comment:
Why the word "upstream"? Since the express purpose here is to remove upstream version checking, I don't quite get it. There aren't a whole lot of packages in the repo that we're even the upstream for, so I think this might just add confusion for end users. I think "distribution release" covers it just fine.